core/server/data/migrations/versions/2.2/3-insert-admin-api-client-role.js

```diff
@@ -0,0 +1,86 @@
+const logging = require('../../../../lib/common/logging');
+const merge = require('lodash/merge');
+const models = require('../../../../models');
+const utils = require('../../../schema/fixtures/utils');
+
+const _private = {};
+
+_private.addApiKeyRole = (options) => {
+    const message = 'Adding "Admin API Client" role to roles table';
+    const apiKeyRole = utils.findModelFixtureEntry('Role', {name: 'Admin API Client'});
+
+    return models.Role.findOne({name: apiKeyRole.name}, options)
+        .then((role) => {
+            if (!role) {
+                logging.info(message);
+                return utils.addFixturesForModel({
+                    name: 'Role',
+                    entries: [apiKeyRole]
+                }, options);
+            }
+
+            logging.warn(message);
+        });
+};
+
+_private.addApiKeyPermissions = (options) => {
+    const message = 'Adding permissions for the "Admin API Client" role';
+    const relations = utils.findRelationFixture('Role', 'Permission');
+
+    return utils.addFixturesForRelation({
+        from: relations.from,
+        to: relations.to,
+        entries: {
+            'Admin API Client': relations.entries['Admin API Client']
+        }
+    }, options).then((result) => {
+        if (result.done === result.expected) {
+            logging.info(message);
+            return;
+        }
+
+        logging.warn(`(${result.done}/${result.expected}) ${message}`);
+    });
+};
+
+_private.removeApiKeyPermissionsAndRole = (options) => {
+    const message = 'Rollback: Removing "Admin API Client" role and permissions';
+
+    return models.Role.findOne({name: 'Admin API Client'}, options)
+        .then((role) => {
+            if (!role) {
+                logging.warn(message);
+                return;
+            }
+
+            return role.related('permissions')
+                .invokeThen('destroy', options)
+                .then(() => role.destroy())
+                .then(() => {
+                    logging.info(message);
```

Comment:
Can we pls put this into the role model?
If i delete a role, the related permissions are getting destroyed. 

`models.Role.destroy(..)`

Model layer takes care to auto-remove relations.

Reply:
Do we have a convention for doing this? I had a quick look and the only place I could see where associated records are cleaned up is when a user is destroyed but that [is handled in the API layer](https://github.com/TryGhost/Ghost/blob/master/core/server/api/users.js#L264-L278)

Reply:
This was discussed in Slack. @kevinansfield Let me know when it's pushed.

Reply:
@kirrg001 pushed the cleanup into the role model with unit test. Added as a separate commit for review but can be squashed